torch/_inductor/runtime/triton_heuristics.py

```diff
@@ -481,12 +481,41 @@ def _precompile_config(self, cfg: Config, warm_cache_only: bool):
                 raise
             binary._init_handles()
 
+        """
+        https://github.com/pytorch/pytorch/issues/115344
+
+        self.fn.constexprs doesn't properly deal with None args, so when we filter out
+        an arg in UserDefinedTritonKernel.codegen, we need to filter it here as well.
+        We also don't want to modify self.fn.
+
+        We know that we removed something from the signature if:
+            1. It's in compile_meta["constants"]
+            2. It isn't a constant we already know about
+                Note: The value of interest has already been added to compile_meta['constants'],
+                    so we use self.fn.constexprs instead.
+            3. It isn't in the compile_meta signature
+        """
+        self.arg_names = self.fn.arg_names
+        removed_from_sig = set(compile_meta["constants"].keys())
```

Comment:
Wondering if we could have a better name for `removed_from_sig`? Otherwise a bit hard to understand what this holds (and why it's usable for both `call_args` and `def_args` below).

Reply:
maybe something like `filtered_none_args`?

Reply:
Yeah, this is more explicit, thanks!